Lecture-14/src/test/java/RestClientTest.java

```diff
@@ -0,0 +1,55 @@
+import constant.LibraryType;
+import factory.ClientFactory;
+import model.GitUser;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import service.ApacheRepositoryService;
+import service.RestAssuredRepositoryService;
+
+import java.util.List;
+
+public class RestClientTest {
+
+    private ApacheRepositoryService apacheRepositoryService;
+    private RestAssuredRepositoryService restAssuredRepositoryService;
+
+    @BeforeClass
+    public void setup() {
+        apacheRepositoryService = new ApacheRepositoryService(ClientFactory.getInstance(LibraryType.APACHE_HTTP));
+        restAssuredRepositoryService = new RestAssuredRepositoryService(ClientFactory.getInstance(LibraryType.REST_ASSURED));
+    }
+
+    @Test
+    public void apacheGetCollaboratorsTest() {
+        System.out.println("\n----------- Apache test ------------");
+        List<GitUser> collaborators = apacheRepositoryService.getCollaborators();
+        System.out.println("collaborators: " + collaborators);
+        Assert.assertEquals(200, apacheRepositoryService.getGitHubResponseFromGet().getStatusCode());
+    }
+
+    @Test
+    public void apacheSendInvitationTest() {
```

Comment:
Use dataprovider to test different implementations

Reply:
done